open-craft/xblock-drag-and-drop-v2
# Account for decoy items in score

## Description

In order to take the correct non-placement of decoy items into account, this PR implements the student score as `score = ((correctly placed items) + (correctly unplaced decoys)) / (non-decoys + decoys)`

## Dependencies

- #87 (now merged)

## JIRA

- [SOL-1988](https://openedx.atlassian.net/browse/SOL-1988)

## Sandbox URLs

- [LMS](http://lms.arbrand.es/courses/course-v1:test+tt101+run/courseware/chapter/drag_and_drop_v2/)
- [Studio](http://cms.arbrand.es/container/block-v1:test+tt101+run+type@vertical+block@drag_and_drop_v2)

## Latest commit

- a1b6931 *(updated 2016-08-18)*

## Testing instructions

1. Go to the [pre-deployed DnDv2 instance](http://lms.arbrand.es/courses/course-v1:test+tt101+run/courseware/chapter/drag_and_drop_v2/) on the LMS, and log in as `staff@example.com`.
2. Click on `Staff Debug Info`, and `Delete Student State` for the `staff@example.com` user.
3. Close the debug info window, drag all items into their correct places, and finally drag the `I don't belong anywhere` item anywhere onto the image.
4. Click `Submit`.
5. Go to the [progress page](http://lms.arbrand.es/courses/course-v1:test+tt101+run/progress), and check that the score for `Drag and Drop v2` is **80%**, or 4 out of 5.

## Screenshots

### Suggested submission

![screenshot](https://cloud.githubusercontent.com/assets/759355/17751446/85f7b13c-649d-11e6-9ef9-fcb78f8922ea.png)

### Expected grade

![screenshot2](https://cloud.githubusercontent.com/assets/759355/17751445/85f792ec-649d-11e6-8038-df15e290f862.png)
